presto-main/src/test/java/com/facebook/presto/sql/planner/TestLogicalPlanner.java

```diff
@@ -278,6 +278,28 @@ public void testCorrelatedSubqueries()
                                                 ))))));
     }
 
+    /**
+     * Handling of correlated IN pulls up everything possible to the generated outer join condition.
+     * This test ensures uncorrelated conditions are pushed back down.
+     */
+    @Test
+    public void testCorrelatedInUncorrelatedFiltersPushDown()
+    {
+        assertPlan(
+                "SELECT orderkey, comment IN (SELECT clerk FROM orders s WHERE s.orderkey = o.orderkey AND s.orderkey < 7) FROM lineitem o",
+                anyTree(
```

Comment:
This is "testCorrelatedInUncorrelatedFiltersPushDown" test. It's not a test for a whole plan, it's just for one aspect of it. 

Here, I feel more confidence for a whole plan by testing queries end to end. Whole-plan test would be quite complex.